receiver/wavefrontreceiver/receiver.go

```diff
@@ -55,7 +55,13 @@ func (r *metricsReceiver) Start(ctx context.Context, host component.Host) error
 		},
 	}
 
-	carbonReceiver, err := fact.CreateMetrics(ctx, r.set, carbonCfg, r.nextConsumer)
+	carbonSettings := receiver.Settings{
+		ID:                component.MustNewIDWithName("carbon", r.set.ID.Name()),
+		TelemetrySettings: r.set.TelemetrySettings,
+		BuildInfo:         r.set.BuildInfo,
+	}
+
+	carbonReceiver, err := fact.CreateMetrics(ctx, carbonSettings, carbonCfg, r.nextConsumer)
```

Comment:
Please do like here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38284, it avoids mistakes to not copy all the fields.